azure-pipelines.yml

```diff
@@ -31,28 +31,49 @@ variables:
   shouldPublish: $[or(eq(variables['Build.SourceBranch'], 'refs/heads/dev'), startsWith(variables['Build.SourceBranch'], 'refs/heads/release/'))]
   ${{ if startsWith(variables['Build.SourceBranch'], 'refs/heads/') }}:
     branchName: $[ replace(variables['Build.SourceBranch'], 'refs/heads/', '') ]
+    branchToCompare: $[ variables['Build.SourceBranch'] ]
   ${{ if startsWith(variables['Build.SourceBranch'], 'refs/pull/') }}:
     branchName: $[ replace(variables['System.PullRequest.SourceBranch'], 'refs/heads/', '') ]
+    branchToCompare: $[ format('origin/{0}{1}', variables['System.PullRequest.TargetBranch'], '...HEAD') ]
+
+# Some steps here scaffold jss apps and check if they build and lint correctly
+# For performance reasons we try to limit the number of packages being built and linted to those that were changed in current branch
+# We use --since param from that to compare current branch to the target - and get only the changes that get pulled
+# We ignore nextjs apps in build command for now - nextjs needs to connect to Sitecore app to be built properly at the moment.
+# create-sitecore-jss gets ignored for lint as it fails with 'The command line is too long.' error.
 
 steps:
 - task: NodeTool@0
   inputs:
     versionSpec: '16.x'
 - script: |
     yarn cache clean --all && yarn install --immutable
-  displayName: 'yarn install'
+  displayName: 'yarn install - initial'
 - script: |
-    yarn build
-  displayName: 'building packages'
+    lerna run lint --since $(branchToCompare)
+  displayName: 'lint packages'
 - script: |
-    yarn lint-packages
-  displayName: 'linting packages'
+    yarn build
+  displayName: 'build packages'
 - script: |
     yarn coverage-packages
   displayName: 'running tests'
   env: {
     testEnv: 'ci'
   }
+- script: |
+    lerna run scaffold-apps
+  displayName: 'scaffold apps'
+- script: |
+    yarn install
+  displayName: 'yarn install - refresh'
+- script: |
+    lerna run lint --ignore create-sitecore-jss --since $(branchToCompare) -- --fix
+  displayName: 'lint apps - updated only'
+- script: |
+    lerna run build --ignore sample-nextjs-* --since $(branchToCompare)
+  displayName: 'build apps - updated only'
```

Comment:
Nothing to do here necessarily, but wanted to put this in writing :)

As we discussed, there will be some duplication of lint/build on the base packages but this is necessary for `--since` resolution of dependent samples/apps. The alternative would be to remove `--since` and use `--scope sample-*` instead, but then we are potentially spending a lot of time on the samples. I think this is an OK trade-off for now?